server/src/main/java/org/opensearch/action/search/SearchResponse.java

```diff
@@ -604,6 +661,118 @@ public String toString() {
         }
     }
 
+    /**
+     * Holds info about the clusters that the search was executed on: how many in total, how many of them were successful
+     * and how many of them were skipped.
+     *
+     * @opensearch.internal
+     */
+    public static class PhaseTook implements ToXContentFragment, Writeable {
+        public static final PhaseTook NULL = new PhaseTook(-1, -1, -1, -1, -1);
+
+        static final ParseField PHASE_TOOK = new ParseField("phase_took");
+        static final ParseField DFS_PREQUERY_FIELD = new ParseField("dfs_prequery");
+        static final ParseField CAN_MATCH_FIELD = new ParseField("can_match");
+        static final ParseField QUERY_FIELD = new ParseField("query");
+        static final ParseField FETCH_FIELD = new ParseField("fetch");
+        static final ParseField EXPAND_FIELD = new ParseField("expand_search");
+
+        private final long dfsPreQueryTotal;
+        private final long canMatchTotal;
+        private final long queryTotal;
+        private final long fetchTotal;
+        private final long expandSearchTotal;
+
+        public PhaseTook(long dfsPreQueryTotal, long canMatchTotal, long queryTotal, long fetchTotal, long expandSearchTotal) {
+            this.dfsPreQueryTotal = dfsPreQueryTotal;
+            this.canMatchTotal = canMatchTotal;
+            this.queryTotal = queryTotal;
+            this.fetchTotal = fetchTotal;
+            this.expandSearchTotal = expandSearchTotal;
+        }
+
+        private PhaseTook(StreamInput in) throws IOException {
+            this(in.readLong(), in.readLong(), in.readLong(), in.readLong(), in.readLong());
```

Comment:
We need to have version guards check around this change as it is a new feature. For now, use 3.0 version. We will eventually change it to 2.11 later on.